find-mmc-bypartlabel

```diff
@@ -20,28 +20,28 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 # Pass the partition label name as $1
-if test -z $1; then
+if [ -z "$1" ]; then
 	echo "$0: Error, no label name given!" > /dev/kmsg
 	exit 1
 fi
 
-if ! test -d /sys/class/block; then
+if [ ! -d /sys/class/block ]; then
 	echo "$0: Error, please mount sysfs" > /dev/kmsg
 	exit 1
 fi
 
-while [ ! -d /sys/class/block/mmcblk0p* ]; do
+while [ ! -d /sys/class/block/mmcblk0p* ] && [ ! -d /sys/class/block/sda ] ; do
     echo "find-mmc-bypartlabel: Waiting for /sys/class/block/mmcblk0p*..." > /dev/kmsg
-    (( i++ ))
+    i=$(( ${i:-0} + 1 ))
     sleep 0.5
-    if [ $i == 10 ]; then
+    if [ $i = 10 ]; then
         echo "find-mmc-bypartlabel: Error: timeout waiting for /sys/class/block/mmcblk0p*" > /dev/kmsg
         exit 1
     fi
 done
 
-for mmc_sysfs in $(ls -d /sys/class/block/mmcblk0p*); do
-	if cat $mmc_sysfs/uevent | grep -w PARTNAME=$1 > /dev/null; then
+for mmc_sysfs in /sys/class/block/mmcblk0p*/*/ /sys/class/block/sda/*/; do
```

Comment:
we need partition names over here: `sd*/*/`

Reply:
But the script ignores everything after mmcblk0* why should it do not for everything after sda*?

Reply:
Fixed that the error was that it was sda/*/ which means all subdir of sda and not sda*/ all dir in this folder beginning with sda.